### DISCUSSION TITLE
Adding polyglot book support in Stockfish

### DISCUSSION THREAD

**Discussion by AlexB123** [official-stockfish/Stockfish#5081] [General]
Hello! I have a problem with adding support for polyglot (.bin) books in the current Stockfish. Before it was easy, was needed just a few tiny changes in code, specifically in Makefile, main.cpp, search.cpp and ucioption.cpp, also need to add 2 extra files polybook.cpp and polybook.h (like in old Brainfish). Anyway, as I was saying everything was fine until this patch "Refactor global variables" https://github.com/official-stockfish/Stockfish/commit/a10791095150bf7c020b92be0f55566fe34e9bf2
Since my programming knowledge is "non-existent", can someone help me please?
Below is the source code of the SF polyglot 080124 "Update default main net to nn-baff1edbea57.nnue" (with changes in files that I mentioned above, polybook.cpp and polybook.h included, perhaps they will need some changes too). Also there is original code of the same SF 080124 (for comparison, just in case). Please take a look in the code and tell me what kind of changes it need.
Thanks in advance.
Regards.
[SF-Polyglot-080124-Update default main net to nn-baff1edbea57.nnue.zip](https://github.com/official-stockfish/Stockfish/files/14445626/SF-Polyglot-080124-Update.default.main.net.to.nn-baff1edbea57.nnue.zip)

**Comment by AlexB123** [comment]
Sadly. No one can help?

**Comment by locutus2** [comment]
I created a branch which incorporates the polybook stuff in the current master: https://github.com/locutus2/Stockfish/tree/polybook
It compiles and loads the book but no warranty for anything (i doesn't use polyglot books).

**Comment by locutus2** [comment]
I also fixed a bug in the old polybook code. The UCI options Book2 BestBookMove was ignored and instead the corrresponding option for the first book was also used for the second book.

**Comment by ArminHHJ** [comment]
Hi,
thanks for the effort, if it works it's a great thing!

What am I doing wrong?
I am downloading your polybook-SF as a zip. Unzip into a PB-Fish directory
Use the following compile call: make -j profile-build ARCH=x86-64-bmi2 COMP=mingw
=> missing polybook files!

I copy the two files from the original polybook-SF into the PB-Fisch directory/src
make -j profile-build ARCH=x86-64-bmi2 COMP=mingw
=> compile aborts with missing "polybook.h"

Are files missing in your new PB-SF?
Or is my procedure not correct?

Armin

PS, master-build compile works in my environment according to my procedure

**Comment by locutus2** [comment]
I have forgot to add the two new polybook files to repo (you can't use the original ones because their sligthly changed). Done. Should now work.

**Reply by AlexB123** [reply]
Hi Stefan, many thanks for your help, much appreciated!!

**Comment by ArminHHJ** [comment]
Great, it works now.
First test copiled fil use the polybook.
Thank you.

Can this PB version use the current master sources in the future?
What would I have to do as a "command"?

**Reply by Disservin** [reply]
> Can this PB version use the current master sources in the future?

Wdym by that? Stockfish will continuously change, and hard coded solutions like these will stop working at some point.

**Comment by ArminHHJ** [comment]
understand. thx :-)

**Comment by winapiadmin** [comment]
stockfish now is (mostly) smart enough for opening that you don't need polyglot books anymore
